src/OrchardCore.Modules/OrchardCore.Liquid/Services/SlugService.cs

```diff
@@ -1,75 +1,70 @@
             }
 
-            var sb = new StringBuilder();
+            var slug = new StringBuilder();
+            var appendHyphen = false;
+            var normalizedText = text.ToLowerInvariant().Normalize(NormalizationForm.FormKD);
 
-            var stFormKD = text.Trim().ToLower().Normalize(NormalizationForm.FormKD);
-            foreach (var t in stFormKD)
+            for (var i = 0; i < normalizedText.Length; i++)
             {
-                // Allowed symbols
-                if (t == '-' || t == '_' || t == '~')
+                var currentChar = normalizedText[i];
+
+                if (CharUnicodeInfo.GetUnicodeCategory(currentChar) == UnicodeCategory.NonSpacingMark)
                 {
-                    sb.Append(t);
                     continue;
                 }
 
-                var uc = CharUnicodeInfo.GetUnicodeCategory(t);
-                switch (uc)
+                if (Char.IsLetterOrDigit(currentChar))
                 {
-                    case UnicodeCategory.LowercaseLetter:
-                    case UnicodeCategory.OtherLetter:
-                    case UnicodeCategory.DecimalDigitNumber:
-                        // Keep letters and digits
-                        sb.Append(t);
-                        break;
-                    case UnicodeCategory.NonSpacingMark:
-                        // Remove diacritics
-                        break;
-                    default:
-                        // Replace all other chars with dash
-                        sb.Append('-');
-                        break;
-                }
-            }
-
-            var slug = sb.ToString().Normalize(NormalizationForm.FormC);
+                    slug.Append(currentChar);
 
-            // Simplifies dash groups
-            for (var i = 0; i < slug.Length - 1; i++)
-            {
-                if (slug[i] == '-')
+                    appendHyphen = true;
+                }
+                else if(_allowedSymbols.Contains(currentChar))
                 {
-                    var j = 0;
-                    while (i + j + 1 < slug.Length && slug[i + j + 1] == '-')
+                    if (currentChar == Hyphen)
                     {
-                        j++;
+                        if (appendHyphen && i != normalizedText.Length - 1)
+                        {
+                            slug.Append(currentChar);
+
+                            appendHyphen = false;
+                        }
                     }
-                    if (j > 0)
+                    else
                     {
-                        slug = slug.Remove(i + 1, j);
+                        slug.Append(currentChar);
                     }
                 }
-            }
+                else
+                {
+                    if (appendHyphen)
+                    {
+                        slug.Append(Hyphen);
 
-            if (slug.Length > 1000)
-            {
-                slug = slug.Substring(0, 1000);
```

Comment:
You appear to have removed this logic?

Reply:
Could you please let me know what's the use of it? Is it to only take 1000 chars if the slug exceed the limits or is there something special